partiql-parser/src/parse/mod.rs

```diff
@@ -372,6 +372,13 @@ mod tests {
             "#;
             parse!(q)
         }
+
+        #[test]
+        fn set_ops() {
+            parse!(
+                r#"SELECT * FROM a LIMIT 10 OFFSET 2 UNION SELECT * FROM b INTERSECT c EXCEPT SELECT * FROM d"#
```

Comment:
Not a blocker: Can we also add a comment on what this equivalent to? If not mistaken it should evaluate to the following?
`SELECT * FROM a LIMIT 10 OFFSET 2 UNION (SELECT * FROM b INTERSECT c EXCEPT SELECT * FROM d)`

Reply:
On a related note, it may be worth adding a TODO to add more operator precedence tests. The Kotlin implementation has some operator precedence tests: https://github.com/partiql/partiql-lang-kotlin/blob/34625c68dbcbaf7b8ae60df7a4cf65c60b2a3b79/lang/test/org/partiql/lang/syntax/SqlParserPrecedenceTest.kt.

I decided not to add them to the conformance test suite at this time since we don't have any ast checks currently.

Reply:
We should look into these tests.

Tying it to the structure of the AST isn't really necessary. 

We should be able to assert the "equality of the AST" of two queries for the purposes of precendence testing.
e.g. `a union b union c` == `(a union b) union c`